### DISCUSSION TITLE
树形表格，如何控制节点动态的展开和收起呢？

### DISCUSSION THREAD

**Discussion by LMGO** [antvis/S2#2283] [Q&A] [answered]
描述：树形表格，如何展开某一行？比如，点击表格外的一个按钮，想通过事件展开具体的某一行，这种能否实现或者如何实现呢？
<img width="391" alt="image" src="https://github.com/antvis/S2/assets/31820043/201964ce-afed-497a-93e6-3a6cc90592e4">

**Comment by lijinke666** [comment, marked as answer]
你看的这个 demo 不就有示例了吗? 表格默认是展开全部, 动态更新 `s2Options.style.collapsedRows` 配置即可

![image](https://github.com/antvis/S2/assets/21015895/4d9f94e0-dfd8-432c-98df-ac115a25235d)

> 如何获取节点 ID: https://s2.antv.antgroup.com/manual/advanced/get-cell-data#%E8%8E%B7%E5%8F%96%E6%8C%87%E5%AE%9A%E5%8C%BA%E5%9F%9F%E5%8D%95%E5%85%83%E6%A0%BC

```ts
s2.setOptions({
  style: {
    // 折叠所有
    hierarchyCollapse: true
    // 展开/折叠某个节点
    collapsedRows: {
      [节点 ID]: false
	}
  }
})

s2.render(false)
```

https://s2.antv.antgroup.com/examples/basic/pivot/#tree

https://s2.antv.antgroup.com/api/general/s2-options#style:~:text=%E4%BB%8E%200%20%E5%BC%80%E5%A7%8B%EF%BC%89%E3%80%82-,collapsedRows,-Record%3Cstring%2C%20boolean

**Reply by LMGO** [reply]
收到，感谢🙏
